.github/workflows/release-snapshot.yml

```diff
@@ -0,0 +1,29 @@
+name: Release Snapshot to Sonatype
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  release_snapshot_sonatype:
+    if: "github.event.release.prerelease"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: corretto
+          java-version: 8
+          cache: sbt
+      - name: Release Snapshot to Sonatype
+        run: |
+          VERSION=$(git describe --tags | cut -f2 -d"@")
+          echo $PGP_SECRET | base64 --decode | gpg --batch --import
+          export GPG_TTY=$(tty)
+          echo "Releasing version $VERSION Sonatype as snapshot"
+          sbt "clean" "project clientCI" "set isSnapshot := true" "release release-version $VERSION with-defaults"
+        env:
+          PGP_SECRET: ${{ secrets.PGP_SECRET }}
```

Comment:
These secrets have been added to the repository. There is no org-wide pgp key or sonatype account, so I've had to use my own personal creds. I don't _think_ there's a better alternative (apart from replacing these with keys from someone in the CAPI team?)